src/map/battle.c

```diff
@@ -4440,7 +4441,7 @@ static struct Damage battle_calc_weapon_attack(struct block_list *src, struct bl
 	)
 		flag.arrow = 1;
 
-	if(skill_id) {
+	if (skill_id || !(skill->get_nk(skill_id)&NK_CRITICAL)) {
```

Comment:
Because here is not bool value, need check for 0 or other constant:

``if (skill_id || (skill->get_nk(skill_id) & NK_CRITICAL) == 0) {``

Reply:
Already change.